doc/glossary.rst

```diff
@@ -1848,26 +1848,56 @@ See concept :term:`sample property`.
-        common tests.
-
-        Some estimators, such as decision trees, support negative weights.
-        FIXME: This feature or its absence may not be tested or documented in
-        many estimators.
-
-        This is not entirely the case where other parameters of the model
-        consider the number of samples in a region, as with ``min_samples`` in
-        :class:`cluster.DBSCAN`.  In this case, a count of samples becomes
-        to a sum of their weights.
-
-        In classification, sample weights can also be specified as a function
-        of class with the :term:`class_weight` estimator :term:`parameter`.
+        A relative weight for each sample. Intuitively, if all weights are
+        integers, using them in a model or scorer is like duplicating each
+        sample as many times as the weight value. Weights may be specified as
+        floats, so that sample weights are usually equivalent up to a constant
+        positive scaling factor.
+
+        `sample_weight` can be both an argument of the estimator's `fit` method
+        for model training or a parameter of a :term:`scorer` for model
+        evaluation. These callables are said to *consume* the sample weights
+        while other components of scikit-learn can *route*  the weights to the
+        underlying estimators or scorers (see
+        :ref:`glossary_metadata_routing`).
+
+        Weighting samples can be useful in several contexts. For instance, if
+        the training data is not uniformly sampled from the target population,
+        it can be corrected by weighting the training data points based on the
+        `inverse probability
+        <https://en.wikipedia.org/wiki/Inverse_probability_weighting>`_ of
+        their selection for training (e.g. inverse propensity weighting). It is
+        also useful to model the frequency of an event of interest per unit of
+        time on a dataset of observations with different exposure durations per
+        individual (see
+        :ref:`sphx_glr_auto_examples_linear_model_plot_poisson_regression_non_normal_loss.py`
+        and
+        :ref:`sphx_glr_auto_examples_linear_model_plot_tweedie_regression_insurance_claims.py`).
+
+        Some model hyper-parameters are expressed in terms of a discrete number
+        of samples in a region of the feature space. When fitting with sample
+        weights, a count of samples is often automatically converted to a sum
+        of their weights, but this is not always the case. Please refer to the
+        docstring for details.
+
+        In classification, weights can also be specified for all samples
+        belonging to a given target class with the :term:`class_weight`
+        estimator :term:`parameter`. If both ``sample_weight`` and
+        ``class_weight`` are provided, the final weight assigned to a sample is
+        the product of the two.
+
+        At the time of writing (version 1.7), not all scikit-learn estimators
```

Comment:
Since 1.7 was released, this doc will be published on scikit-learn.org/stable no sooner than 1.8.

```suggestion
        At the time of writing (version 1.8), not all scikit-learn estimators
```